File: challenge37.py
import sys

def is_pangram(line):
    """Returns the required letters to create a pangram or NULL if it is one."""
    req_letters = [chr(i) for i in xrange(97, 123)]

    letters = list(line.strip().lower())
    for letter in letters:
        # Ensure the letter is in a-z set
        if ord(letter) >= 97 and ord(letter) <= 122:
            # Remove it from required letters
            if letter in req_letters:
                req_letters.remove(letter)

    if req_letters == []:
        return 'NULL'
    return ''.join(req_letters)

test_cases = open(sys.argv[1], 'r')

for test in test_cases:
    print is_pangram(test)

test_cases.close()

File: Ruby/challenge37.rb
def is_pangram(line)
  # Returns letter missing from line to create a pangram or NULL if it is one
  req_letters = (97..122).map(&:chr)

  line.downcase.each_char { |c|
    if c.ord >= 97 && c.ord <= 122
      if req_letters.include?(c)
        req_letters.delete(c)
      end
    end
  }
  return req_letters == [] ? 'NULL' : req_letters.join
end

File.open(ARGV[0]).each_line do |line|
  puts is_pangram(line.strip)
end
